tests/cloudpickle_test.py

```diff
@@ -508,6 +509,21 @@ def test_load_dynamic_module_in_grandchild_process(self):
             if os.path.exists(child_process_module_file):
                 os.unlink(child_process_module_file)
 
+    def test_correct_globals_import(self):
+        import math
+
+        def my_small_function(x, y):
+            return x+y
+
+        b = cloudpickle.dumps(my_small_function)
+
+        # Cloudpickle will use the current module's name for base_globals.
+        # Separately, global variables used by my_small_function will be
+        # pickled in f_globals. So at no point should 'math' appear in the
+        # pickled object, since it is not used by my_small_function explicitly
+        assert b'my_small_function' in b
+        assert b'math' not in b
```

Comment:
Check that `unwanted_function` is not in `b` while `nested_function` is there.

Reply:
This is for non-regression purpose. If we ever introduce a change that fail this test, it will be detected with such test.

Reply:
OK. Adding this part then.